src/record-error-archival/src/DatabaseClient.ts

```diff
@@ -0,0 +1,121 @@
+import { Client } from "pg"
+import { logger } from "shared"
+import type { PromiseResult } from "shared-types"
+
+export type ArchivedErrorRecord = {
+  messageId: string
+  errorId: number
+  archivedAt: Date
+  archivedBy: string
+  archiveLogId: number
+}
+
+export type DatabaseRow = {
+  message_id: string
+  error_id: number
+  archived_at: string
+  archived_by: string
+  archive_log_id: number
+}
+
+// Produces the string "$1, $2, $3..." for the given range
+const wherePlaceholderForRange = (range: number) => [...Array(range).keys()].map((x) => `$${x + 1}`).join(", ")
+
+export default class DatabaseClient {
+  private postgres: Client
+
+  constructor(
+    private host: string,
+    private user: string,
+    private password: string,
+    private useSsl: boolean,
+    private database: string,
+    private schema: string,
+    private archiveGroupLimit: number
+  ) {
+    this.postgres = new Client({
+      host: this.host,
+      user: this.user,
+      password: this.password,
+      ssl: this.useSsl,
```

Comment:
If SSL is true then we need to return `{ rejectUnauthorized: false }`, otherwise it won't work.
```
ssl: config.useSsl ? { rejectUnauthorized: false } : false
```